### PR TITLE
chore(envbuilder-dogfood): update envbuilder-dogfood template to use provider

### PR DESCRIPTION
**Pull request by johnstcn**
Updates the `envbuilder-dogfood` template to use the `envbuilder` provider.

Relates to coder/team-coconut#38

### CODE REVIEW

File: envbuilder-dogfood/main.tf [hunk 1]
```diff
@@ -336,33 +339,49 @@ resource "docker_image" "envbuilder" {
...
-    "ENVBUILDER_DOCKER_CONFIG_BASE64=${data.local_sensitive_file.envbuilder_cache_dockerconfigjson.content_base64}",
-    "USE_CAP_NET_ADMIN=true",
-    # Set git commit details correctly
-    "GIT_AUTHOR_NAME=${coalesce(data.coder_workspace_owner.me.full_name, data.coder_workspace_owner.me.name)}",
-    "GIT_AUTHOR_EMAIL=${data.coder_workspace_owner.me.email}",
-    "GIT_COMMITTER_NAME=${coalesce(data.coder_workspace_owner.me.full_name, data.coder_workspace_owner.me.name)}",
-    "GIT_COMMITTER_EMAIL=${data.coder_workspace_owner.me.email}",
-  ]
+  # Use environment computed from the provider
+  env = envbuilder_cached_image.cached.0.env
```
Comment (matifali) [inline envbuilder-dogfood/main.tf:384]: **Not related to this PR**:
Can we also access individual envs from `envbuilder_cached_image.cached.0.env`?

Comment (johnstcn) [reply envbuilder-dogfood/main.tf:384]: I added `env_map` to make it easier. 
https://registry.terraform.io/providers/coder/envbuilder/latest/docs/resources/cached_image#env_map

File: envbuilder-dogfood/main.tf [hunk 2]
```diff
@@ -336,33 +339,49 @@ resource "docker_image" "envbuilder" {
   keep_locally  = true
 }
 
+locals {
+  cache_repo = "us-central1-docker.pkg.dev/coder-dogfood-v2/envbuilder-cache/coder-dogfood"
+  envbuilder_env = {
+    "CODER_AGENT_TOKEN" : coder_agent.dev.token,
+    "CODER_AGENT_URL" : data.coder_workspace.me.access_url,
```
Comment (mafredri) [inline envbuilder-dogfood/main.tf:346]: I suppose these could be set on the container only, but fine to keep here for convenience 👍🏻

File: envbuilder-dogfood/main.tf [hunk 3]
```diff
@@ -336,33 +339,49 @@ resource "docker_image" "envbuilder" {
...
+    "ENVBUILDER_GIT_USERNAME" : data.coder_external_auth.github.access_token,
+    # "ENVBUILDER_GIT_URL" : data.coder_parameter.devcontainer_repo.value, # This is set automatically by the provider.
```
Comment (mafredri) [inline envbuilder-dogfood/main.tf:348]:
```suggestion
    # "ENVBUILDER_GIT_URL" : data.coder_parameter.devcontainer_repo.value, # The provider sets this via the `git_url` property.
```

File: envbuilder-dogfood/main.tf [hunk 4]
```diff
@@ -336,33 +339,49 @@ resource "docker_image" "envbuilder" {
...
+    "ENVBUILDER_DEVCONTAINER_DIR" : data.coder_parameter.devcontainer_dir.value,
+    "ENVBUILDER_INIT_SCRIPT" : coder_agent.dev.init_script,
+    "ENVBUILDER_FALLBACK_IMAGE" : "codercom/oss-dogfood:latest", # This image runs if builds fail
+    "ENVBUILDER_PUSH_IMAGE" : "true",                            # Push the image to the remote cache
+    # "ENVBUILDER_CACHE_REPO" : local.cache_repo, # This is set automatically by the provider.
```
Comment (mafredri) [inline envbuilder-dogfood/main.tf:353]:
```suggestion
    # "ENVBUILDER_CACHE_REPO" : local.cache_repo, # The provider sets this via the `cache_repo` property.
```

**Review by mafredri — APPROVED** [review]
Nice!

**Pull request merged**